.github/workflows/hil_test_linux.yml

```diff
@@ -25,9 +25,40 @@ on:
         type: string
 
 jobs:
+  hil-matrix:
+    name: linux-hil-matrix
+    runs-on: ubuntu-latest
+
+    outputs:
+      tests: ${{ steps.output-tests.outputs.tests }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Prepare tests matrix
+        id: output-tests
+        shell: python
```

Comment:
Yeah, neither did I. But since we had already used `shell: bash` and I wanted a bit more (string manipulation) logic that I was comfortable to implement in bash, I figured I will give it a try :) I think it is very cool, especially in the context implemented here, which would be tricky in bash. So... I would probably execute Python from within bash if Python would not be supported directly.

Reply:
Yeah, I agree, really helpful. Just confusing at first when I started reading the script assuming it was bash haha.